### PR TITLE
Fix IfThenElse in CudaCodeGen. Add unique func name

### PR DESCRIPTION
**Pull request by zheng-xq**

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp [hunk 1]
```diff
@@ -372,6 +383,18 @@ class PrioritizeLoad : public IRMutator {
   }
 
   MemoryLoadStack load_stack_;
+  // TODO: For now, we are not movign the loads with the IfThenElse.
```
Comment (bertmaher) [inline torch/csrc/jit/tensorexpr/cuda_codegen.cpp]: sp: "moving"

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp [hunk 2]
```diff
@@ -253,17 +253,21 @@ void CudaPrinter::visit(const LetStmt* v) {
...
   os() << " : ";
   v->false_value()->accept(this);
+  os() << ")";
 }
 
 class PrioritizeLoad : public IRMutator {
  public:
-  virtual const Expr* mutate(const Load* v) {
+  const Expr* mutate(const Load* v) override {
+    if (nested_if_then_else_ > 0) {
```
Comment (bertmaher) [inline torch/csrc/jit/tensorexpr/cuda_codegen.cpp:269]: You might add a comment here that points ahead to the bigger comment at the variable declaration.

**Review by bertmaher — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp (hunk 1 above)
Comment (zheng-xq) [reply torch/csrc/jit/tensorexpr/cuda_codegen.cpp]: Done

File: torch/csrc/jit/tensorexpr/cuda_codegen.cpp (hunk 2 above)
Comment (zheng-xq) [reply torch/csrc/jit/tensorexpr/cuda_codegen.cpp:269]: Done

**Pull request merged**